File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    answer = ''
    numbers = list(map(str, numbers))
    numbers.sort(key = lambda x : x*3, reverse=True)
    answer = str(int(answer.join(numbers)))
    return answer

# 각 숫자를 세 번 반복하여 비교하고, 그 결과를 기준으로 정렬합니다. 이렇게 함으로써 숫자의 크기를 비교할 때, 숫자의 자릿수를 고려한 정렬이 가능해집니다.

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.Arrays;

class Solution {
        public String solution(int[] numbers) {
            String answer = "";
            StringBuilder sb = new StringBuilder();
            String[] answerLst = new String[numbers.length];
            for (int i=0; i<numbers.length; i++) {
                answerLst[i] = String.valueOf(numbers[i]);
            }

            Arrays.sort(answerLst, (n1, n2) -> (n2+n1).compareTo(n1+n2)); // 두 숫자를 문자열로 붙인 결과를 비교하여 정렬

            if (answerLst[0].equals("0")) return "0"; // 첫번째 요소가 0이면, 모든 요소가 0이므로 최종적으로 0

            for (String number : answerLst) sb.append(number);
            answer = String.valueOf(sb);
            return answer;
        }
}